sdks/python/apache_beam/io/iobase.py

```diff
@@ -1013,3 +1013,56 @@ def process(self, element):
     if self.counter >= self.count:
       self.counter -= self.count
     yield self.counter, element
+
+
+class RestrictionTracker(object):
+  """Manages concurrent access to a restriction.
+
+  Experimental; no backwards-compatibility guarantees.
+
+  Keeps track of the restrictions claimed part for a Splittable DoFn.
+
+  See following documents for more details.
+  * https://s.apache.org/splittable-do-fn
+  * https://s.apache.org/splittable-do-fn-python-sdk
+  """
+
+  def current_restriction(self):
+    """Returns the current restriction.
+
+    Returns a restriction accurately describing the full range of work the
+    current ``DoFn.process()`` call will do, including already completed work.
+    """
+    raise NotImplementedError
+
+  def checkpoint(self):
```

Comment:
<!--thread_id:cc_141415269_t; commit:95a209dccc3398be93d21bd9d926a886343b9617; resolved:0-->
<!--section:context-quote-->
> **robertwb** wrote:
> We shouldn't introduce a method that we know will be redundant with future developments.

<!--section:body-->
Seems like, based on previous discussions [1], we are moving towards maintaining both methods due to the need for calling checkpoint() after a resume() ? Also Java API currently has the checkpoint() method, we should add this to Python for consistency and if needed update both APIs (again maintaining consistency) after future discussions/experiments ? 

[1] https://docs.google.com/document/d/1BGc8pM1GOvZhwR9SARSVte-20XEoBUxrGJ5gTWXdv3c/edit
[2] https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/RestrictionTracker.java#L42

Reply:
Semantically, checkpoint() enforces that immediately following claim (try_claim() once we have that method) should be rejected while try_split(0) will not enforce that.